### PR TITLE
remove suffixes from generated StorageClasses and VolumeSnapshotClass

### LINKED ISSUE #96234: Clean up / remove suffixes from generated StorageClasses and VolumeSnapshotClasses in E2E tests
**What would you like to be added**:
Suffixes for generated `StorageClass` and `VolumeSnapshotClass` objects can be removed since they already have randomly generated names.
https://github.com/kubernetes/kubernetes/blob/396b90f06ca29c754a81eb6c03cff50b7d9bf265/test/e2e/storage/testsuites/driveroperations.go#L76

**Why is this needed**:
This is a small cleanup to remove an argument that isn't needed (it's hardcoded in callers) and / or repeats the name of the snapshotter in the suffix which can reduce readability:
https://github.com/kubernetes/kubernetes/blob/396b90f06ca29c754a81eb6c03cff50b7d9bf265/test/e2e/storage/drivers/csi.go#L163-L170

/sig storage

### PR DESCRIPTION
**Pull request by lala123912**
What type of PR is this?
/kind cleanup

What this PR does / why we need it:
Remove redundant suffix to improve readability.
Which issue(s) this PR fixes:
Fixes #96234

Special notes for your reviewer:

Does this PR introduce a user-facing change?:
NONE

Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

### CODE REVIEW

**Comment by lala123912** [comment]
this is privious pr https://github.com/kubernetes/kubernetes/pull/96373, I meet some mistakes when rebasing the commits, so I raising a new one.

**Comment by chrishenzie** [comment]
/lgtm
/approve

Thank you!

File: test/e2e/storage/testsuites/driveroperations.go [hunk 1]
```diff
@@ -107,7 +105,7 @@ func GetSnapshotClass(
 			"metadata": map[string]interface{}{
 				// Name must be unique, so let's base it on namespace name and use GenerateName
 				// TODO(#96234): Remove unnecessary suffix.
-				"name": names.SimpleNameGenerator.GenerateName(ns + "-" + suffix),
+				"name": names.SimpleNameGenerator.GenerateName(ns),
```
Comment (pacoxu) [inline test/e2e/storage/testsuites/driveroperations.go:108]: > // TODO(#96234): Remove unnecessary suffix.

Remove the comment as well ?

Comment (lala123912) [reply test/e2e/storage/testsuites/driveroperations.go:106]: > > https://github.com/kubernetes/kubernetes/pull/97183/files#diff-f18ed9fa9279e79977b207010ee6601f97178ed4a5720610bd67275ef5a50ce3R107
> > // TODO(#96234): Remove unnecessary suffix.
> 
> Remove the comment as well ?

Ok, I have removed it and updated the code. @pacoxu

**Review by pacoxu — APPROVED** [review]

**Comment by chrishenzie** [comment]
/lgtm

Thank you!

**Pull request merged**